src/script/auth/module/action/AuthAction.ts

```diff
@@ -91,6 +98,25 @@ export class AuthAction {
     };
   };
 
+  /**
+   * Temporary solution to be used ONLY with non-persistent temporary clients. It's a workaround to switch between
+   * Dexie (IndexedDB) and SQLeetEngine (encrypted IndexedDB). When we fully use SQLeetEngine we can move this
+   * configuration to `configureClient`.
+   */
+  private async initEncryptedDatabase(): Promise<SQLeetEngine> {
+    const existingKey: string = await getEphemeralValue();
+    const encryptionKey = existingKey || (await saveRandomEncryptionKey());
+    return StorageService.initEncryptedDatabase(encryptionKey);
+  }
+
+  doFlushDatabase = (): ThunkAction => {
+    return async (dispatch, getState, {apiClient}) => {
+      if (apiClient.config.store instanceof SQLeetEngine) {
+        apiClient.config.store.save();
```

Comment:
```suggestion
        await apiClient.config.store.save();
```

We should await here as well and within the sqleet-engine (which we're currently not).